OpenCabConsumer/app/src/main/java/com/eleostech/exampleconsumer/MainActivity.java

```diff
@@ -79,56 +77,161 @@ public void onDestroy() {
 
     public void onEvent(VehicleInformationChangedEvent event) {
         Log.d(LOG_TAG, "onVehicleInformationChanged()");
-        clear();
+        logEvent(event.action);
     }
 
-    private void clear() {
-        gearView.setText("");
-        vinView.setText("");
-        idView.setText("");
-        errorView.setText("");
+    public void onEvent(IdentityChangedEvent event) {
+        Log.d(LOG_TAG, "IdentityChangedEvent()");
+        logEvent(event.action);
     }
 
-    private  Date addHoursToDate(Date date, int hours) {
-        Calendar calendar = Calendar.getInstance();
-        calendar.setTime(date);
-        calendar.add(Calendar.HOUR_OF_DAY, hours);
-        return calendar.getTime();
+    private void logEvent(String event) {
+        SimpleDateFormat s = new SimpleDateFormat("MM/dd hh:mm:ss");
+        String dateTime = s.format(new Date());
+        adapterBroadcastedEvents.insert(dateTime + " : " + event, 0);
     }
 
-    private void callProvider() {
-        Log.d(LOG_TAG, "callProvider()");
-        clear();
-        ContentResolver resolver = getApplicationContext().getContentResolver();
-        Uri authority = Uri.parse("content://" + VehicleInformationContract.AUTHORITY);
-        Bundle result = null;
-        try {
-            result = resolver.call(authority, "getVehicleInformation", "0.2", null);
-        } catch (Exception ex) {
-            Log.i(LOG_TAG, "Error calling provider: ", ex);
-            errorView.setText(ex.getMessage());
-            return;
+    private void callVehicleInformationProvider() {
+        Log.d(LOG_TAG, "callVehicleInformationProvider()");
+        List<PackageInfo> packages = getPackageManager().getInstalledPackages(PackageManager.GET_PROVIDERS);
+        search:
+        for (PackageInfo pkg : packages) {
+            if (pkg.providers != null) {
+                for (ProviderInfo provider : pkg.providers) {
+                    if (provider.authority != null) {
+                        if (provider.authority.endsWith(".org.opencabstandard.vehicleinformation")) {
```

Comment:
For other reviewers, this is essentially the key change: provider apps are now enumerated and located based on a specific authority _suffix_ rather than being registered at a global authority string. This allows multiple provider apps to be installed at once, since authorities must be unique across all installed apps.